capture/plugins/writer-s3.c

```diff
@@ -261,9 +262,38 @@ void writer_s3_request(char *method, char *path, char *qs, unsigned char *data,
     g_checksum_reset(checksum);
     g_checksum_update(checksum, data, len);
     strcpy(bodyHash, g_checksum_get_string(checksum));
-    snprintf(canonicalRequest, sizeof(canonicalRequest),
+    if (s3PathAccessStyle) {
```

Comment:
Minor nit.  Instead of this big if, could you have just a fullpath var like below and use that in this url?

Reply:
> Minor nit. Instead of this big if, could you have just a fullpath var like below and use that in this url?

I couldn't reuse the fullpath var as it also had the querystring appended to it, so I added a new var. Not ideal but works and looks nicer :)